lib/functions/compilation/uboot.sh

```diff
@@ -194,9 +200,10 @@ function compile_uboot_target() {
 
 	display_alert "${uboot_prefix}Compiling u-boot" "${version} ${target_make} with gcc '${gcc_version_main}'" "info"
 	declare -g if_error_detail_message="${uboot_prefix}Failed to build u-boot ${version} ${target_make}"
+	# FIXME-QA(Krey): I don't like the handling of OPENSBI here
 	do_with_ccache_statistics run_host_command_logged_long_running \
 		"CFLAGS='${uboot_cflags}'" "KCFLAGS='${uboot_cflags}'" \
-		CCACHE_BASEDIR="$(pwd)" PATH="${toolchain}:${toolchain2}:${PATH}" \
+		CCACHE_BASEDIR="$(pwd)" OPENSBI="$([ -z "$OPENSBISOURCE" ] || echo "$SRC/cache/sources/opensbi/fw_dynamic.bin")" PATH="${toolchain}:${toolchain2}:${PATH}" \
```

Comment:
I don't like the handling here, but i couldn't think of a way to improve it without rewriting things, kept as is for easy review